wire-ios-sync-engine/Source/Synchronization/Strategies/CallingRequestStrategy.swift

```diff
@@ -267,38 +269,40 @@ public final class CallingRequestStrategy: AbstractRequestStrategy, ZMSingleRequ
         }
     }
 
-    func processCallEvent(conversationUUID: UUID,
-                          senderUUID: UUID,
-                          clientId: String,
-                          conversationDomain: String?,
-                          senderDomain: String?,
-                          payload: Data,
-                          currentTimestamp: TimeInterval,
-                          eventTimestamp: Date) {
-        let conversationId = AVSIdentifier(
-            identifier: conversationUUID,
-            domain: conversationDomain
-        )
-        let userId = AVSIdentifier(
-            identifier: senderUUID,
-            domain: senderDomain
-        )
-
-        let callEvent = CallEvent(
-            data: payload,
-            currentTimestamp: Date().addingTimeInterval(currentTimestamp),
-            serverTimestamp: eventTimestamp,
-            conversationId: conversationId,
-            userId: userId,
-            clientId: clientId
-        )
-
-        callEventStatus.scheduledCallEventForProcessing()
-
-        callCenter?.processCallEvent(callEvent, completionHandler: { [weak self] in
-            self?.zmLog.debug("processed calling message")
-            self?.callEventStatus.finishedProcessingCallEvent()
-        })
+    func processCallEvent(
+        callingConversationId: QualifiedConversationId,
+        conversationUUID: UUID,
+        senderUUID: UUID,
+        clientId: String,
+        conversationDomain: String?,
+        senderDomain: String?,
+        payload: Data,
+        currentTimestamp: TimeInterval,
+        eventTimestamp: Date) {
+            let conversationId = AVSIdentifier(
+                identifier: !callingConversationId.id.isEmpty ? UUID(uuidString: callingConversationId.id)! : conversationUUID,
+                domain: !callingConversationId.domain.isEmpty ? callingConversationId.domain : conversationDomain
+            )
```

Comment:
Formatting suggestion. Note: suggested indentation is not right
```suggestion
        let identifier = !callingConversationId.id.isEmpty
            ? UUID(uuidString: callingConversationId.id)!
            : conversationUUID

        let domain = !callingConversationId.domain.isEmpty
            ? callingConversationId.domain
            : conversationDomain

        let conversationId = AVSIdentifier(
            identifier: identifier,
            domain: domain
        )
```

Reply:
updated